MapboxCoreNavigation/PassiveLocationDataSource.swift

```diff
@@ -0,0 +1,169 @@
+import MapboxNavigationNative
+import CoreLocation
+import MapboxDirections
+import MapboxAccounts
+
+/**
+ An object that notifies its delegate when the user’s location changes, minimizing the noise that normally accompanies location updates from a `CLLocationManager` object.
+ 
+ Unlike `Router` classes such as `RouteController` and `LegacyRouteController`, this class operates without a predefined route, matching the user’s location to the road network at large. You can use a passive location manager to determine a starting point for a route that you calculate using the `Directions.calculate(_:completionHandler:)` method. If the user happens to be moving while you calculate the route, the passive location manager makes it less likely that the route will begin with a short segment on a side road or driveway and a confusing instruction to turn onto the current road.
+ 
+ To find out when the user’s location changes, implement the `PassiveLocationDataSourceDelegate` protocol, or observe `Notification.Name.passiveLocationDataSourceDidUpdate` notifications for more detailed information.
+ */
+open class PassiveLocationDataSource: NSObject {
+    /**
+     Initializes the location data source with the given directions service.
+     
+     - parameter directions: The directions service that allows the location data source to access road network data. If this argument is omitted, the shared `Directions` object is used.
+     
+     - postcondition: Call `startUpdatingLocation(completionHandler:)` afterwards to begin receiving location updates.
+     */
+    public required init(directions: Directions = Directions.shared, systemLocationManager: NavigationLocationManager? = nil) {
+        self.directions = directions
+        
+        let settingsProfile = SettingsProfile(application: ProfileApplication.kMobile, platform: ProfilePlatform.KIOS)
+        self.navigator = Navigator(profile: settingsProfile, config: NavigatorConfig() , customConfig: "")
```

Comment:
I remember that @SiarheiFedartsou was mentioning that it can be beneficial to use one global `MBNavigator` and feed it with data whenever it's available. Since we now have 4 (?) places where `MBNavigator` is created it might make sense to consider one global instance approach.

Reply:
Right, for better or worse, the navigator is intended to be used as something of a singleton if I remember correctly.

#2509 will remove two of these initializations, leaving just the ones in RouteController and PassiveLocationDataSource, so I didn’t bother to centralize the logic just yet.

The long-term vision is indeed to unify PassiveLocationDataSource and RouteController, so that the latter is just a special case of setting a route on the former and keeping track of route progress. For now, we would probably recommend that the developer stop location updates on PassiveLocationDataSource when beginning turn-by-turn navigation with RouteController. This happens to be the case in the example application, because it demonstrates temporarily taking the main map view out of the view hierarchy as a performance optimization.